packages/react/src/generators/application/application.spec.ts

```diff
@@ -19,6 +18,17 @@ import { Schema } from './schema';
 //  which is v9 while we are testing for the new v10 version
 jest.mock('@nx/cypress/src/utils/cypress-version');
 
+let projectGraph: ProjectGraph;
+jest.mock('@nx/devkit', () => {
```

Comment:
Rather than read the package from graph, use `readModulePackageJson` from `nx/src/devkit-internals` to find the version of React.

Reply:
@Coly010 brought up a point of the function not supporting Yarn PnP, and also that the root package.json not always containing `react`. The graph contains the deduped version so it will always exist.

Reply:
The graph provides a more accurate source of truth for package versions across different package managers and package manager resolution types (bun and yarn pnp where no node_modules folder exists).

Let's stick to reading graph.